lnwallet/btcwallet/btcwallet_test.go

```diff
@@ -0,0 +1,168 @@
+package btcwallet
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/wallet"
+)
+
+type previousOutpoint struct {
+	outpoint    string
+	isOurOutput bool
+}
+
+type previousOutpointsTest struct {
+	name     string
+	tx       *wire.MsgTx
+	myInputs []wallet.TransactionSummaryInput
+	expRes   []previousOutpoint
+}
+
+var previousOutpointsTests = []previousOutpointsTest{
+	{
+		name: "both outpoints are wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
+			{
+				Index: 0,
+			},
+			{
+				Index: 1,
+			},
+		},
+		expRes: []previousOutpoint{
+			{
+				outpoint: wire.OutPoint{
+					Index: 0,
+				}.String(),
+				isOurOutput: true,
+			},
+			{
+				outpoint: wire.OutPoint{
+					Index: 1,
+				}.String(),
+				isOurOutput: true,
+			},
+		},
+	},
+	{
+		name: "only one outpoint is wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
+			{
+				Index: 0,
+			},
+			{
+				Index: 2,
+			},
+		},
+		expRes: []previousOutpoint{
+			{
+				outpoint: wire.OutPoint{
+					Index: 0,
+				}.String(),
+				isOurOutput: true,
+			},
+			{
+				outpoint: wire.OutPoint{
+					Index: 1,
+				}.String(),
+				isOurOutput: false,
+			},
+		},
+	},
+	{
+		name: "no outpoint is wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
```

Comment:
Maybe also add a test case with an empty `TxIn` and one with an empty `myInputs` slice.